### PR TITLE
Filter inactive items with Switch "hide" button

### PR DESCRIPTION
**Pull request by hobbitronics**
- Added Switch
- simplified filtering of items
- used "hide inactive items" terminology for users
Screenshot from 2022-05-21 15-50-44

### CODE REVIEW

**Comment by hobbitronics** [comment]
I had to merge in the feature branch that updated to ui-components@6 in order to get this to build since Switch must rely on a later version of material.

File: src/components/mdc/Switch/Switch.svelte [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
-    switchControl = new MDCSwitch(el)
-  }
+  switchControl = new MDCSwitch(element)
 })
 
 const dispatch = createEventDispatcher()
 
-function onClick () {
-  dispatch('click', switchControl.selected)
+// Used setTimeout because switchControl.selected needs to be updated before dispatching the event
```
Comment (Schparky) [inline src/components/mdc/Switch/Switch.svelte:17]: It's good that you put this comment here.

**Review by Schparky — APPROVED** [review]

**Comment by forevermatt** [comment]
> I had to merge in the feature branch that updated to ui-components@6 in order to get this to build since Switch must rely on a later version of material.

@hobbitronics - Do you want to go ahead and merge that PR to develop first, then? That way this PR will be more clearly just the changes that are part of the new Switch behavior.

File: src/pages/policies/[policyId].svelte [hunk 2]
```diff
@@ -231,7 +226,13 @@ th {
 
   <div class="flex justify-between align-items-center">
     <h4>Items <span class="subtext">({approvedItems?.length} covered)</span></h4>
-    <Button on:click={toggleAllItems}>{showAllItems ? 'Active items...' : 'all items…'}</Button>
+    <div>
+      {#if isAboveTablet()}
+        <Checkbox label="Hide Inactive" on:checked={hideInactiveItems} on:unchecked={showInactiveItems} />
+      {:else}
+        <Switch on:selected={hideInactiveItems} on:deselected={showInactiveItems} label="Hide Inactive off/on" />
```
Comment (forevermatt) [inline src/pages/policies/[policyId].svelte:233]: I think we could probably remove "off/on" from the label:
```suggestion
        <Switch on:selected={hideInactiveItems} on:deselected={showInactiveItems} label="Hide Inactive" />
```

File: src/pages/policies/[policyId].svelte [hunk 3]
```diff
@@ -49,8 +41,8 @@ $: members = policy.members || []
 
 $: policyId && loadItems(policyId)
 // sort items so inactive is last
-$: items = $selectedPolicyItems.filter(showApprovedOnly ? itemIsApproved : itemIsNotInactive)
-$: itemsForTable = showAllItems ? $selectedPolicyItems : items.slice(0, 15)
+$: items = $selectedPolicyItems.slice(0, 15)
```
Comment (forevermatt) [inline src/pages/policies/[policyId].svelte:44]: You may want to wait to slice down to just 15 items until after (optionally) filtering down to approved items, to maintain a consistent number of items in the table.

File: src/pages/policies/[policyId].svelte [hunk 4]
```diff
@@ -49,8 +41,8 @@ $: members = policy.members || []
...
+$: itemsForTable = hideInactive ? items.filter(itemIsApproved) : items
```
Comment (forevermatt) [inline src/pages/policies/[policyId].svelte:45]: Do you want to just use the `approvedItems` variable from the next line (probably pull it above this one, if so)?
```suggestion
$: itemsForTable = hideInactive ? approvedItems : items
```

**Review by forevermatt — APPROVED** [review]
Made a few suggestions, none blocking

File: src/components/mdc/Switch/Switch.svelte (hunk 1 above)
Comment (hobbitronics) [reply src/components/mdc/Switch/Switch.svelte:17]: I had wanted to do something like ```$: dispatch(switchControl.selected ? 'selected' : 'deselected')``` but for some reason the reactive statement wasn't picking up the change, so I settled on this.

**Comment by hobbitronics** [comment]
Thanks for the good feedback @forevermatt and @Schparky

File: src/pages/policies/[policyId].svelte [hunk 5]
```diff
@@ -247,8 +247,8 @@ th {
       on:batchDelete={() => deleteItems(checkedItemIds, policyId)}
     />
     <div class="text-align-center">
-      <p class="item-footer">Showing {itemsForTable.length} out of {$selectedPolicyItems.length} items</p>
-      <Button url={itemsRoute(policyId)}>View {$selectedPolicyItems.length - itemsForTable.length} more items…</Button>
+      <p class="item-footer">Showing {itemsForTable.length} out of {items.length} items</p>
+      <Button url={itemsRoute(policyId)}>View {items.length - itemsForTable.length} more items…</Button>
```
Comment (forevermatt) [inline src/pages/policies/[policyId].svelte:251]: As an unrelated side note, if there are no items we probably don't need to show the button. (I see "View 0 more items..." when I look at that page, which feels a little odd 🙂.)

**Comment by forevermatt** [comment]
> Thanks for the good feedback @forevermatt and @Schparky

Nice work on your part 👍

**Pull request merged**